tests/assets.spec.ts

```diff
@@ -0,0 +1,190 @@
+import { expect, use } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import assets_constructor from '../types/constructors/asset_wrapper_mock';
+import assets_contract from '../types/contracts/asset_wrapper_mock';
+import { ApiPromise, WsProvider, Keyring } from '@polkadot/api';
+import { KeyringPair } from '@polkadot/keyring/types';
+import {buildTx} from "./helper";
+import {afterEach} from "mocha";
+import {WeightV2} from "@polkadot/types/interfaces";
+import {stringToHex} from "@polkadot/util/string/toHex";
+
+use(chaiAsPromised);
+
+// Create a new instance of contract
+const wsProvider = new WsProvider('ws://127.0.0.1:9944');
+// Create a keyring instance
+const keyring = new Keyring({ type: 'sr25519' });
+
+describe('ASSETS', () => {
+    let api: ApiPromise;
+    let alice: KeyringPair;
+    let bob: KeyringPair;
+    let charlie: KeyringPair;
+    let assetsConstructor: assets_constructor
+    let assets: assets_contract;
+
+    let gasRequired: WeightV2;
+
+    beforeEach(async function() {
+        api = await ApiPromise.create({ provider: wsProvider });
+        alice = keyring.addFromUri('//Alice');
+        bob = keyring.addFromUri('//Bob');
+        charlie = keyring.addFromUri('//Charlie');
+        assetsConstructor = new assets_constructor(api, alice);
+        assets = new assets_contract((await assetsConstructor.new()).address, alice, api);
+    })
+
+    afterEach( async function() {
+        await buildTx(api.registry, api.tx.assets.destroy(1, {accounts: 10, sufficients: 10, approvals: 10}), alice)
+    })
+
+    it('create works', async () => {
+        let { gasRequired: gas }  = await assets.query.create(1,1, {
+            value: 1000000,
+        });
+        await assets.tx.create(1, 1,{gasLimit: gas, value: 1000000 });
+
+        // @ts-ignore
+        await expect((await api.query.assets.asset(1)).unwrapOrDefault().owner.toU8a().toString()).to.equal(alice.addressRaw.toString())
+    })
+
+    it('mint works', async () => {
+        let { gasRequired: gas }  = await assets.query.create(1,1, {
+            value: 1000000,
+        });
+        await assets.tx.create(1, 1,{gasLimit: gas, value: 1000000 });
+
+        let { gasRequired: gas2 }  = await assets.query.mint(1,1000);
+        await assets.tx.mint(1, 1000,{gasLimit: gas2 });
+
+        // @ts-ignore
+        await expect((await api.query.assets.account(1, alice.address)).unwrapOrDefault().balance.toNumber()).to.equal(1000)
+    })
+
+    it('burn works', async () => {
+        let { gasRequired: gas }  = await assets.query.create(1,1, {
+            value: 1000000,
+        });
+        await assets.tx.create(1, 1,{gasLimit: gas, value: 1000000 });
+
+        let { gasRequired: gas2 }  = await assets.query.mint(1,1000);
+        await assets.tx.mint(1, 1000,{gasLimit: gas2 });
+
+        let { gasRequired: gas3 }  = await assets.query.burn(1,100);
+        await assets.tx.burn(1, 100,{gasLimit: gas3 });
```

Comment:
who is account 100? How this will be deducted from `who`when that account has no tokens minted?
Again maybe use of account name/alias would make it clear